giskard/models/base/model.py

```diff
@@ -593,3 +601,154 @@ def __str__(self) -> str:
+
+        Parameters
+        ----------
+        dataset : Dataset
+            Giskard Dataset to be analysed by Tools.
+        scan_report : ScanReport
+            Giskard Scan Report to be analysed by Tools.
+
+        Returns
+        -------
+        dict[str, BaseTool]
+            The dictionary with Tools' names and related instances.
+        """
+        try:
+            from ...llm.talk.tools import (
+                IssuesScannerTool,
+                MetricTool,
+                PredictTool,
+                SHAPExplanationTool,
+            )
+        except ImportError as err:
+            raise GiskardInstallationError(flavor="talk") from err
+
+        return {
+            PredictTool.default_name: PredictTool(model=self, dataset=dataset),
+            MetricTool.default_name: MetricTool(model=self, dataset=dataset),
+            SHAPExplanationTool.default_name: SHAPExplanationTool(model=self, dataset=dataset),
+            IssuesScannerTool.default_name: IssuesScannerTool(scan_report=scan_report),
+        }
+
+    @staticmethod
+    def _gather_context(message_list: list) -> str:
+        """Gather context into a single string.
+
+        Given the list of OpenAI's messages, extracts and joins their contents into a single string.
+
+        Parameters
+        ----------
+        message_list : list
+            The list of OpenAI's messages.
+
+        Returns
+        -------
+        str
+            The string with the joined messages' contents.
+        """
+        return "\n".join([json.dumps(asdict(msg)) for msg in message_list])
+
+    def talk(self, question: str, dataset: Dataset, scan_report: ScanReport = None, context: str = ""):
+        """Perform the 'talk' to the model.
+
+        Given `question`, allows to ask the model about prediction result, explanation, model performance, issues, etc.
+
+        Parameters
+        ----------
+        question : str
+            User input query.
+        dataset : Dataset
+            Giskard Dataset to be analysed by the 'talk'.
+        context : str
+            Context of the previous 'talk' results. Necessary to keep context between sequential 'talk' calls.
+        scan_report : ScanReport
+            Giskard Scan Report to be analysed by the 'talk'.
+
+        Returns
+        -------
+        TalkResult
+            The response for the user's prompt.
+        """
+        try:
+            from ...llm import get_copilot_client, set_llm_model
+            from ...llm.client import ToolChatMessage
+            from ...llm.talk.config import (
+                ERROR_RESPONSE,
+                MODEL_INSTRUCTION,
+                SUMMARY_PROMPT,
+                TALK_CLIENT_CONFIG,
+                get_talk_llm_model,
+            )
+            from ..talk_result import TalkResult
+        except ImportError as err:
+            raise GiskardInstallationError(flavor="talk") from err
+
+        set_llm_model(get_talk_llm_model())
```

Comment:
Do we really want to change the default llm model here? It means that if we run other parts of the library (e.g. the scan) after talking to a model, the default llm could be changed.

Reply:
you're right, let me change this